### ISSUE TITLE
error for inference 

### ISSUE THREAD

**Issue by EhsanRusta** [dickreuter/neuron_poker#73]
I'm running the main.py with command: python main.py selfplay random --render
and getting the following error, 
"AssertionError: The environment must specify an observation space. "
I'm new to the gym and reinforcement learning so it may there is a simple task I missed

**Comment by davehensley** [comment]
Unfortunately the recent versions of [gym](https://github.com/openai/gym) broke backward compatibility, and the requirements for this project don't specify a specific version to use.

Try editing the requirements.txt file and changing the line
```
gym
```
to
```
gym==0.23.1
```
then re-run `pip install -r requirements.txt`

**Comment by jaimeggb** [comment]
Hi @davehensley 

I tried what you said and it works, thank you!

However, a new issue appears which requires also adding 

`GLU`

to the requirements.txt file.

Once that is done, there is further progress, but another issue appears which may have something to do with the fact I'm using jupyter notebook on Google Cloud Platform. This is the issue

> (neuron_poker) jupyter@tensorflow-2-8-20221018-113241:~/neuron_poker$ python main.py selfplay keypress --render
/home/jupyter/neuron_poker/tools/helper.py:11: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
  from collections import Iterable  # pylint: disable=no-name-in-module
Using default log file
Saving log file to: /home/jupyter/neuron_poker/log/default.log
Saving info file to: /home/jupyter/neuron_poker/log/default_info.log
Saving error only file to: /home/jupyter/neuron_poker/log/default_errors.log
Screenloglevel: 20
INFO - Initializing program
INFO - 
INFO - ++++++++++++++++++
INFO - Starting new hand.
INFO - ++++++++++++++++++
INFO - Dealer is at position 0
INFO - Player 0 got ['AS', 'JH'] and $500
INFO - Player 1 got ['JD', '2D'] and $500
INFO - 
INFO - ===Round: Stage: PREFLOP
INFO - Seat 1 (Keypress): Action.SMALL_BLIND - Remaining stack: 499, Round pot: 1, Community pot: 0, player pot: 1
INFO - Seat 0 (Keypress): Action.BIG_BLIND - Remaining stack: 498, Round pot: 3, Community pot: 0, player pot: 2
Traceback (most recent call last):
  File "/opt/conda/envs/neuron_poker/lib/python3.7/site-packages/pyglet/__init__.py", line 336, in __getattr__
    return getattr(self._module, name)
AttributeError: 'NoneType' object has no attribute 'Window'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "main.py", line 255, in <module>
    command_line_parser()
  File "main.py", line 68, in command_line_parser
    runner.key_press_agents()
  File "main.py", line 124, in key_press_agents
    self.env.reset()
  File "/opt/conda/envs/neuron_poker/lib/python3.7/site-packages/gym/wrappers/order_enforcing.py", line 18, in reset
    return self.env.reset(**kwargs)
  File "/home/jupyter/neuron_poker/gym_env/env.py", line 178, in reset
    self._get_environment()
  File "/home/jupyter/neuron_poker/gym_env/env.py", line 293, in _get_environment
    self.render()
  File "/home/jupyter/neuron_poker/gym_env/env.py", line 675, in render
    self.viewer = PygletWindow(screen_width + 50, screen_height + 50)
  File "/home/jupyter/neuron_poker/gym_env/rendering.py", line 23, in __init__
    self.display_surface = pyglet.window.Window(width=X, height=Y + 50)
  File "/opt/conda/envs/neuron_poker/lib/python3.7/site-packages/pyglet/__init__.py", line 342, in __getattr__
    __import__(import_name)
  File "/opt/conda/envs/neuron_poker/lib/python3.7/site-packages/pyglet/window/__init__.py", line 1919, in <module>
    gl._create_shadow_window()
  File "/opt/conda/envs/neuron_poker/lib/python3.7/site-packages/pyglet/gl/__init__.py", line 206, in _create_shadow_window
    _shadow_window = Window(width=1, height=1, visible=False)
  File "/opt/conda/envs/neuron_poker/lib/python3.7/site-packages/pyglet/window/xlib/__init__.py", line 171, in __init__
    super(XlibWindow, self).__init__(*args, **kwargs)
  File "/opt/conda/envs/neuron_poker/lib/python3.7/site-packages/pyglet/window/__init__.py", line 591, in __init__
    display = pyglet.canvas.get_display()
  File "/opt/conda/envs/neuron_poker/lib/python3.7/site-packages/pyglet/canvas/__init__.py", line 94, in get_display
    return Display()
  File "/opt/conda/envs/neuron_poker/lib/python3.7/site-packages/pyglet/canvas/xlib.py", line 123, in __init__
    raise NoSuchDisplayException('Cannot connect to "%s"' % name)
pyglet.canvas.xlib.NoSuchDisplayException: Cannot connect to "None"

What could I do?

**Comment by dickreuter** [comment]
fixed
